apps/astarte_appengine_api/lib/astarte_appengine_api/config/non_negative_integer.ex

```diff
@@ -0,0 +1,43 @@
+#
+# This file is part of Astarte.
+#
+# Copyright 2020 Ispirata Srl
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+defmodule Astarte.AppEngine.API.Config.NonNegativeInteger do
+  use Skogsra.Type
+
+  @impl Skogsra.Type
+  @spec cast(String.t()) :: {:ok, integer()} | :error
+  def cast(value)
+
+  def cast(""), do: :error
+
+  def cast(value) when is_binary(value) do
+    with {limit, _} <- Integer.parse(value) do
+      if limit > 0 do
```

Comment:
I would write `limit >= 0` just to underline that the first if branch handles allowed integers, while the other one forces any other to 0.